gandalf.js

```diff
@@ -6,113 +6,123 @@
  *
```

Comment:
SCRIPT-You did a great job handling all the possible error states in the UI. It enhances the user experience significantly.